activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -94,7 +94,7 @@ def logger
       def debug(progname = nil, &block)
         return unless super
 
-        if ActiveRecord::Base.verbose_query_logs
+        if defined?(Rails::Server) && ActiveRecord::Base.verbose_query_logs
```

Comment:
Another thought: sometimes it's really useful to _conditionally_ enable verbose logging in tests (and, maybe, console and other tasks too), i.e. manually turn it on and off; this hack make it impossible.

Moreover, ActiveRecord could be used without Rails at all.

Reply:
Active Record should not know anything about railties. Maybe we should move this code to the railtie?

```ruby
initializer 'set_verbose_logs' do
  ActiveRecord::Base.verbose_query_logs = defined?(Rails::Server) && config.active_record.verbose_query_logs
end
```

Reply:
I think `defined?(Rails::Server)` is still the wrong approach, even if we move it to the Railtie. It will disable the feature in too many places: standalone web servers (`puma`), background job servers (`sidekiq`), even other built-in Rails contexts (`bin/rails runner`)...

We should disable it when the console is used, not enable it when the server is used. I like @palkan's suggestion of using the `console do` block to do that.

Reply:
Can we accomplish this by checking whether the FD that the logger is outputting to is interactive? (basically `out.repond_to?(:tty?) && out.tty?`)

Reply:
I don't think so, because both the server and console commands multiplex their log output to an interactive stream ([STDOUT for the server](https://github.com/rails/rails/blob/f8afb5168aed810a061d081d3941c9dc9bd633c3/railties/lib/rails/commands/server/server_command.rb#L96) and [STERR for the console](https://github.com/rails/rails/blob/f8afb5168aed810a061d081d3941c9dc9bd633c3/activerecord/lib/active_record/railtie.rb#L60)), but we only want to disable verbose query logs on the console.

Ideally we'd always write the verbose query logs to the log file and exclude them from STDOUT/STDERR, but I don't think that's possible with how the multiplexing is implemented.

Reply:
Let's use the `console` block for now in the railtie. If we have another case where we need it disabled too we can revisit.

Reply:
@rafaelfranca Updating the PR with your suggestion now.